tests/test_main.py

```diff
@@ -0,0 +1,206 @@
+import os
+import random
+import string
+from unittest.mock import Mock
+from uuid import uuid4
+
+import responses
+from notifications_python_client import __version__
+
+from main import notify, send_email
+
+url = "https://api.notifications.service.gov.uk/v2/notifications/email"
+success_json = {
+    "id": "740e5834-3a29-46b4-9a6f-16142fde533a",
+    "reference": "STRING",
+    "content": {
+        "subject": "SUBJECT TEXT",
+        "body": "MESSAGE TEXT",
+        "from_email": "SENDER EMAIL",
+    },
+    "uri": "https://api.notifications.service.gov.uk/v2/notifications/740e5834-3a29-46b4-9a6f-16142fde533a",
+    "template": {
+        "id": "f33517ff-2a88-4f6e-b855-c550268ce08a",
+        "version": __version__,
+        "uri": "https://api.notifications.service.gov.uk/v2/template/f33517ff-2a88-4f6e-b855-c550268ce08a",
+    },
+}
+
+
+def test_get_not_allowed():
+    request = Mock(method="GET")
+    response = notify(request)
+    assert response == ("Method not allowed", 405)
+
+
+def test_missing_data_returns_422():
+    request = Mock(method="POST", json={})
+    response = notify(request)
+    assert response == ("Missing notification request data", 422)
+
+
+def test_send_email_invalid_service_id():
+    random_string = "".join(random.choice(string.printable) for i in range(87))
+    result = send_email(random_string, {}, os.getenv("NOTIFY_TEST_TEMPLATE_ID"))
+    assert result == ("Service ID is not a valid uuid", 422)
+
+
+def test_send_email_invalid_api_key():
+    random_string = "".join(random.choice(string.printable) for i in range(37))
+    uuid_string = str(uuid4())
+    result = send_email(
+        uuid_string + random_string, {}, os.getenv("NOTIFY_TEST_TEMPLATE_ID")
+    )
+    assert result == ("API key is not a valid uuid", 422)
+
+
+@responses.activate
+def test_notify_response_error_returns_correctly():
+    request = Mock(
+        method="POST",
+        json={
+            "fulfilmentRequest": {
+                "email_address": "test@example.com",
+                "personalisation": {"address": "test address"},
+                "form_type": "HH",
+                "language_code": "en",
+                "region_code": "Eng",
```

Comment:
This should be updated to a valid region code (applies to all occurrences of this).

Reply:
Done